### PR TITLE
Change g_indent into ntabs parameter of Dump(unsigned int ntabs)

### PR DESCRIPTION
**Pull request by LGM-Doyle**
#### Problem ####
Multiple threads increment/decrement `g_indent` without any synchonization.  

I think that this leads to rare buggy behavior every few thousand turns.  It causes data corruption, log corruption and/or crashes.

#### This Solution ####
Convert `g_indent` from a global that requires synchronization to a parameter of `Dump()`

### CODE REVIEW

File: universe/Building.cpp [hunk 1]
```diff
@@ -228,34 +228,34 @@ std::string BuildingType::Dump() const {
...
 
     if (m_location) {
-        retval += DumpIndent() + "location = \n";
-        ++g_indent;
-            retval += m_location->Dump();
-        --g_indent;
+        retval += DumpIndent(ntabs) + "location = \n";
+        ++ntabs;
+            retval += m_location->Dump(ntabs);
+        --ntabs;
```
Comment (dbenage-cx) [inline universe/Building.cpp]: These simple instances of increment, pass, decrement could now be reduced to `->Dump(ntabs + 1)`

**Review by dbenage-cx — APPROVED** [review]
LGTM, consider comment optional.

File: universe/Building.cpp (hunk 1 above)
Comment (LGM-Doyle) [reply universe/Building.cpp]: Good point.  I made the changes.

**Comment by geoffthemedio** [comment]
Looks reasonable after skimming. OK assuming it's been tested at least a bit.

**Pull request merged**